File: packages/parser-tree-sitter/bindings/go/binding_test.go
// Copyright (c) 2026, Salesforce, Inc.
// All rights reserved.
// SPDX-License-Identifier: Apache-2.0
// For full license text, see the LICENSE file in the repo root or https://www.apache.org/licenses/LICENSE-2.0

package tree_sitter_agentscript_test

import (
	"testing"

	tree_sitter "github.com/tree-sitter/go-tree-sitter"
	tree_sitter_agentscript "github.com/tree-sitter/tree-sitter-agent_script_awl/bindings/go"
)

func TestCanLoadGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_agentscript.Language())
	if language == nil {
		t.Errorf("Error loading AgentScript grammar")
	}
}

File: packages/parser-tree-sitter/bindings/node/binding_test.js
/*
 * Copyright (c) 2026, Salesforce, Inc.
 * All rights reserved.
 * SPDX-License-Identifier: Apache-2.0
 * For full license text, see the LICENSE file in the repo root or https://www.apache.org/licenses/LICENSE-2.0
 */

import assert from 'node:assert';
import { test } from 'node:test';
import Parser from 'tree-sitter';

test('can load grammar', async () => {
  const parser = new Parser();
  await assert.doesNotReject(async () => {
    const { default: language } = await import('./index.js');
    parser.setLanguage(language);
  });
});
